### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by toshipp**
add CONTRIBUTING.md based on CNCF template and modify README.md to adapt to CNCF template.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    ```console
+    cd e2e
+    make start-lvmd
+    make test
+    ```
+
+## Sign Your Commits
+
+### DCO
```
Comment (toshipp) [inline CONTRIBUTING.md:104]: There are two options DCO and CLA. I choose DCO because it is lightweight.

Please provide your though, maintainers.

Comment (cupnes) [reply CONTRIBUTING.md:104]: I choose DCO too. I think CLA is high operating costs and we don't need as much rigor as the CLA.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+## Pull Request Checklist
+
+When you submit your pull request, or you push new commits to it, our automated
+systems will run some checks on your new code. We require that your pull request
+passes these checks, but we also have more criteria than just that before we can
+accept and merge it. We recommend that you check the following things locally
+before you submit your code:
+
+- If your code has breaking change, please update related documents.
```
Comment (cupnes) [inline CONTRIBUTING.md:136]:
```suggestion
- If your code has a breaking change, please update related documents.
```

File: README.md [hunk 3]
```diff
@@ -27,8 +22,30 @@ Supported environments
 
 *1 The host's Linux Kernel must be v4.9 or later which supports `rmapbt` and `reflink`, if you use xfs filesystem with an official docker image.
 
-Features
---------
+Docker images are available on [ghcr.io](https://github.com/orgs/topolvm/packages)
```
Comment (cupnes) [inline README.md:25]:
```suggestion
Docker images are available on [ghcr.io](https://github.com/orgs/topolvm/packages).  
```

File: README.md [hunk 4]
```diff
@@ -42,48 +59,29 @@ Features
...
-Getting started
----------------
+- TopoLVM currently uses [Pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/), but it has been deprecated since Kubernetes 1.21, and will be deleted in Kubernetes 1.25. So it is also deprecated in TopoLVM.
 
-A demonstration of TopoLVM running on [kind (Kubernetes IN Docker)][kind] is available at [example](example/) directory.
+Communications
+--------------
 
-For production deployments, see [deploy/README.md](./deploy/README.md).
+If you have any questions or idea to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).
```
Comment (cupnes) [inline README.md:69]:
```suggestion
If you have any questions or ideas to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).
```

**Review by cupnes — CHANGES_REQUESTED** [review]
Please check and correct them.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
+# Contributing Guide
+
+* [New Contributor Guide](#contributing-guide)
```
Comment (llamerada-jp) [inline CONTRIBUTING.md:3]:
```suggestion
* [Contributing Guide](#contributing-guide)
```
Were those generated automatically? Only this line is different from actual title.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+We welcome many different types of contributions including:
+
+* New features
+* Builds, CI/CD
+* Bug fixes
+* Documentation
+* Issue Triage
+* Answering questions on Github Discussions
+* Web design
```
Comment (llamerada-jp) [inline CONTRIBUTING.md:34]: This one seems to be a different kind. Is it correct?

**Review by llamerada-jp — APPROVED** [review]
Before creating a PR that includes big changes, I would like to discuss it or make a proposal. What do you think?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+If your change is approved by us, we will merge it immediately and it will be
+shipped on the next monthly release.
+
+When there has been no activity for 30 days, the stale bot will label it stale.
+And if there are no activity for another 7 days, it will be closed.
+
+## Development Environment Setup
+
+Our recommended environment is Ubuntu 20.04. Following steps modify your system globally,
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:76]: `Following steps` is a subject.

File: CONTRIBUTING.md (hunk 6 above)
Comment (toshipp) [reply CONTRIBUTING.md:34]: It is contained in the template, so I don't know what it means. It is possibly for documents.
But, we have no web site now, so I delete it.

**Comment by toshipp** [comment]
>Before creating a PR that includes big changes, I would like to discuss it or make a proposal. What do you think?

Yeah, it is better. I will improve it.

**Review by llamerada-jp — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -27,8 +22,30 @@ Supported environments
...
+
+User manual is at [docs/user-manual.md](docs/user-manual.md).
+
+_Deprecated: If you want to use TopoLVM on [Rancher/RKE](https://rancher.com/docs/rke/latest/en/), see [docs/deprecated/rancher.md](docs/deprecated/rancher.md)._
+
+Contributing
+------------
+
+Our project welcomes contributions from any member of our community. To get
+started contributing, please see our [Contributor Guide](CONTRIBUTING.md).
```
Comment (peng225) [inline README.md:43]:
```suggestion
started contributing, please see our [Contributing Guide](CONTRIBUTING.md).
```

**Review by peng225 — CHANGES_REQUESTED** [review]

**Comment by peng225** [comment]
For other reviewers:
https://github.com/cncf/project-template/blob/main/CONTRIBUTING.md

**Review by cupnes — APPROVED** [review]

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+* Release management
+
+Not everything happens through a GitHub pull request. Please [contact us](https://github.com/topolvm/topolvm/discussions)
+and let's discuss how we can work together.
+
+## Find an Issue
+
+We have good first issues for new contributors and help wanted issues suitable
+for any contributor. [good first issue](https://github.com/topolvm/topolvm/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) has extra information to
+help you make your first contribution. [help wanted](https://github.com/topolvm/topolvm/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+label%3A%22help+wanted%22) are issues
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:44]: It is a mistake. Thanks.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+* The original github issue
+
+## Pull Request Lifecycle
+
+If you want to make a non-trivial change, such as a new feature or a bug fix to break exiting behavior,
+please discuss it first with maintainers on a issue, discussion or a proposal PR you send.
+
+When you submit a PR, it is not a draft and all tests pass, we will start
+a review process. Reviewers are selected automatically, so you do not care about them.
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:68]: The original template uses both... (e.g. isn't and have not).
I don't know the rule, but change do + not to don't and is + not to isn't.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+2. Install the required tools.
+
+    ```console
+    cd topolvm
+    make setup
+    ```
+
+3. Make changes you wish.
+
+4. Test your changes by end-to-end test.
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:96]: Indeed.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+The best way to reach us with a question when contributing is to ask on:
+
+* The original github issue
+
+## Pull Request Lifecycle
+
+If you want to make a non-trivial change, such as a new feature or a bug fix to break exiting behavior,
+please discuss it first with maintainers on a issue, discussion or a proposal PR you send.
+
+When you submit a PR, it is not a draft and all tests pass, we will start
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:67]: fixed.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+If your change is approved by us, we will merge it immediately and it will be
+shipped on the next monthly release.
+
+When there has been no activity for 30 days, the stale bot will label it stale.
+And if there is no activity for another 7 days, it will be closed.
+
+## Development Environment Setup
+
+Our recommended environment is Ubuntu 20.04. Following steps modify your system globally,
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:78]: fixed.

File: README.md (hunk 8 above)
Comment (toshipp) [reply README.md:43]: fixed. (it is from the template though :)

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+issue to work on the [Discussions](https://github.com/topolvm/topolvm/discussions).
+
+Once you see an issue that you'd like to work on, please post a comment saying
+that you want to work on it. Something like "I want to work on this" is fine.
+
+## Ask for Help
+
+The best way to reach us with a question when contributing is to ask on:
+
+* The original github issue
```
[the remark this answers is not in the document]
Comment (toshipp) [reply CONTRIBUTING.md:60]: I interpreted it as that if you have questions to work on the issue, what is it the best way to reach us? I think it is on the original issue.
By contrast, it is preferable to use discussions to ask general questions including how to contribute, IMO.

[the remark this answers is not in the document]
Comment (peng225) [reply CONTRIBUTING.md:60]: Obeying the CNCF's template, I withdraw this comment.
https://github.com/cncf/project-template/blob/main/CONTRIBUTING.md#ask-for-help

**Review by peng225 — DISMISSED** [review]

**Comment by toshipp** [comment]
I'll squash this after you approve. Please don't merge this.

**Review by daichimukai — CHANGES_REQUESTED** [review]
LGTM. Block merge until squash.

**Comment by daichimukai** [comment]
When squash commits, please "sign off" on commits!

File: README.md [hunk 15]
```diff
@@ -42,48 +59,29 @@ Features
...
-- `topolvm-scheduler`: A [scheduler extender](https://github.com/kubernetes/design-proposals-archive/blob/main/scheduling/scheduler_extender.md) for TopoLVM.
-- `topolvm-node`: CSI node service.
-- `lvmd`: gRPC service to manage LVM volumes.
-
-`lvmd` is a standalone program that should be run on Node OS as a systemd service.
-Other programs are packaged into [a container image](https://github.com/orgs/topolvm/packages).
-
-Getting started
----------------
+- TopoLVM currently uses [Pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/), but it has been deprecated since Kubernetes 1.21, and will be deleted in Kubernetes 1.25. So it is also deprecated in TopoLVM.
```
Comment (satoru-takeuchi) [inline README.md:64]:
```suggestion
- [Pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/) support is deprecated and will be removed when TopoLVM removes the support of Kubernetes < v1.25. This policy is same as Kubernetes.
```

File: README.md [hunk 16]
```diff
@@ -42,48 +59,29 @@ Features
...
 
-For production deployments, see [deploy/README.md](./deploy/README.md).
+If you have any questions or ideas to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).
 
-User manual is at [docs/user-manual.md](docs/user-manual.md).
+Resources
+---------
 
-_Deprecated: If you want to use TopoLVM on [Rancher/RKE](https://rancher.com/docs/rke/latest/en/), see [docs/deprecated/rancher.md](docs/deprecated/rancher.md)._
+[docs](docs/) directory contains documents about designs and specifications.
```
Comment (satoru-takeuchi) [inline README.md:74]:
```suggestion
[docs](docs/) directory contains the user manual, designs and specifications, and so on.
```

File: README.md [hunk 17]
```diff
@@ -42,48 +59,29 @@ Features
...
 
-Documentation
--------------
+A diagram of components is available in [docs/design.md](docs/design.md#diagram).
 
-[docs](docs/) directory contains documents about designs and specifications.
+The team presented the motivation and implementation of TopoLVM at KubeCon Europe 2020: https://kccnceu20.sched.com/event/ZerD
```
Comment (satoru-takeuchi) [inline README.md:78]:
```suggestion
TopoLVM maintainers presented the motivation and implementation of TopoLVM at KubeCon Europe 2020: https://kccnceu20.sched.com/event/ZerD
```

File: README.md [hunk 18]
```diff
@@ -10,15 +10,10 @@ TopoLVM
...
-The team presented the motivation and implementation of TopoLVM at KubeCon Europe 2020: https://kccnceu20.sched.com/event/ZerD
-
-If you have any questions, please use [discussions](https://github.com/topolvm/topolvm/discussions).
-
 - **Project Status**: Testing for production
 - **Conformed CSI version**: [1.5.0](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md)
 
-Supported environments
-----------------------
+Our supported platform is:
```
[the remark this answers is not in the document]
Comment (toshipp) [reply README.md:16]: done

File: README.md [hunk 19]
```diff
@@ -27,8 +22,30 @@ Supported environments
 
 *1 The host's Linux Kernel must be v4.9 or later which supports `rmapbt` and `reflink`, if you use xfs filesystem with an official docker image.
 
-Features
---------
+Docker images are available on [ghcr.io](https://github.com/orgs/topolvm/packages).  
+**Please note that [the images on quay.io](https://quay.io/organization/topolvm) are currently deprecated and will no longer be available in the future.**
```
[the remark this answers is not in the document]
Comment (toshipp) [reply README.md:26]: done

File: README.md [hunk 20]
```diff
@@ -27,8 +22,30 @@ Supported environments
...
+For production deployments, see [deploy/README.md](./deploy/README.md).
+
+User manual is at [docs/user-manual.md](docs/user-manual.md).
+
+_Deprecated: If you want to use TopoLVM on [Rancher/RKE](https://rancher.com/docs/rke/latest/en/), see [docs/deprecated/rancher.md](docs/deprecated/rancher.md)._
+
+Contributing
+------------
+
+Our project welcomes contributions from any member of our community. To get
```
[the remark this answers is not in the document]
Comment (toshipp) [reply README.md:42]: done

File: README.md [hunk 21]
```diff
@@ -42,48 +59,29 @@ Features
...
-Getting started
----------------
+- TopoLVM currently uses [Pod security policy](https://kubernetes.io/docs/concepts/policy/pod-security-policy/), but it has been deprecated since Kubernetes 1.21, and will be deleted in Kubernetes 1.25. So it is also deprecated in TopoLVM.
 
-A demonstration of TopoLVM running on [kind (Kubernetes IN Docker)][kind] is available at [example](example/) directory.
+Communications
+--------------
 
-For production deployments, see [deploy/README.md](./deploy/README.md).
+If you have any questions or ideas to contribute, please use [discussions](https://github.com/topolvm/topolvm/discussions).
```
[the remark this answers is not in the document]
Comment (toshipp) [reply README.md:69]: done

File: README.md (hunk 15 above)
Comment (toshipp) [reply README.md:64]: done

File: README.md (hunk 16 above)
Comment (toshipp) [reply README.md:74]: done

File: README.md (hunk 17 above)
Comment (toshipp) [reply README.md:78]: done

**Review by satoru-takeuchi — APPROVED** [review]

**Review by cupnes — APPROVED** [review]

**Comment by toshipp** [comment]
squashed. please approve it again.

**Review by cupnes — APPROVED** [review]

**Review by daichimukai — APPROVED** [review]

**Review by peng225 — APPROVED** [review]

**Pull request merged**